doc/wg/network/notes/network-notes-2024-09-23.md

```diff
@@ -0,0 +1,90 @@
+ * Tyler: I am thinking maybe WiFi and VNC is okay?
+ * Leon: Or maybe "Guacamole" works on the RPi, so people could just use a browser tab
+ * Tyler: We might just warn people that they need a laptop that they have dev privilege on
+ * Leon: We could buy precisely identical RPis for tutorials that we are using for CI anyways
+ * Leon: Or for times when latency isn't an issue, we could literally use Treadmill
+ * Tyler: Okay, action items are to circle back with Microsoft folks and concretely deciding what our plan to try out is for dev environment
+
+
+## 15.4 Integration Testing
+ * Tyler: What's the state of Treadmill? Where's it at and what are next steps for me to work on a 15.4 test?
+ * Leon: First, there's a PR for integrating Treadmill into CI as of today! https://github.com/tock/tock/pull/4177
+ * Leon: State now. First runs a Github actions job which spawns a variety of jobs for Treadmill to handle dynamically which actually run on different boards. Those jobs can be programmed with Github Actions. You write instructions like you do for any other Github workflow, but every shell command is running on the Treadmill RPis.
+ * Leon: Right now, all it does is compile the board and does an lsusb to check that the board is connected.
+ * Leon: In the next PR, we'll add a Python script with some basic integration tests.
+ * Leon: So mapping to 15.4, we probably want to run sometimes, but not on every PR. Maybe nightly and let it run for an hour or two. And I think Tyler has scripts which already run stuff.
+ * Leon: One thing that's missing right now: don't have a way to guarantee that when you launch a jobset of multiple jobs on multiple boards, you don't have an assurance that they'll run in parallel. So if we have three RPis, each of the three can have jobs, but the current form doesn't guarantee that they _will_ run in parallel.
+ * Leon: The way to solve this now is that you can reserve all three of these, and only launch jobs on the reservation itself. It's possible that another job sneaks in though.
+ * Leon: So, an integration test across distributed boards on distributed hosts is tough right now
+ * Tyler: So reservations: we could have three RPis here at UCSD which are only for networks?
+ * Leon: Well, we don't really have reservations, but we have permissions. So only one user could put stuff on some RPis. Which works about the same
+ * Tyler: And my scripts just rely on the serial output to see if packets are received. There were some messages on the Matrix channel about how to do that. Are tests currently using serial output?
+ * Tyler: We don't have a working proposal or consensus right now. Ben is actively working on it. We really want to separate the testing workloads from the Treadmill platform. Treadmill can run arbitrary commands. We will want a standard way to do integration tests, but we haven't fully decided that. And 15.4 testing could have a different interface if it needs
```

Comment:
```suggestion
 * Leon: We don't have a working proposal or consensus right now. Ben is actively working on it. We really want to separate the testing workloads from the Treadmill platform. Treadmill can run arbitrary commands. We will want a standard way to do integration tests, but we haven't fully decided that. And 15.4 testing could have a different interface if it needs
```

Reply:
That does sound like something Leon would say...